plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -40,11 +41,12 @@ var ErrNoNodesAvailable = fmt.Errorf("no nodes available to schedule pods")
 
 // implementation of the error interface
 func (f *FitError) Error() string {
-	output := fmt.Sprintf("failed to find fit for pod, ")
+	predicates := util.NewStringSet()
 	for node, predicateList := range f.FailedPredicates {
-		output = output + fmt.Sprintf("Node %s: %s", node, strings.Join(predicateList.List(), ","))
+		predicates = predicates.Union(predicateList)
+		glog.Infof("failed to find fit for pod %v on node %s: %s", f.Pod.Name, node, strings.Join(predicateList.List(), ","))
 	}
-	return output
+	return fmt.Sprintf("Pod %v failed at least one of the following fitness predicates on every node: %v.", f.Pod.Name, strings.Join(predicates.List(), ","))
```

Comment:
This wording is a little ambiguous. (One wrong interpretation is that there is some fitness predicate in the list that the pod failed on every node, which isn't true.) I would say "For each of these fitness predicates, pod XXX failed on at least one node: ..."  which is slightly better